### PR TITLE
EA Giving Portal UI

### PR DESCRIPTION
**Pull request by oetherington**
This PR has the basic UI for the giving season giving portal ([Figma](https://www.figma.com/file/3WOTZHH10ZBXSZPWw9omM3/Giving-Portal?node-id=25%3A15487&mode=dev)).

I think the best thing to do is to submit this as a PR as-is, and then we can start a feature branch for all the giving season features that this can be merged into.

There's currently no backend changes here - all the orgs, etc. are just hard-coded mock data that can be swapped out for backend API calls in a future PR. The buttons also currently don't do anything.

The "classic writing" section is just pull the top inflation-adjusted posts of all time tagged with effective giving.

The donation election tag doesn't exist yet so, for now, that section is also just showing effective giving posts but this will be a one line change once the tag is created.

The timeline is currently rendered to-scale which makes it a bit more cramped than in the mockups - it might be that we don't want to do this(?).

Screenshot 2023-10-19 at 14 19 54
Screenshot 2023-10-19 at 14 20 04
Screenshot 2023-10-19 at 14 20 12

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205752609339135) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
@agnesstenlund There's a couple of things you'll want to have a look at here:

- I added some basic colors for dark mode just to make things readable, but it's definitely not pretty. I'm sure you come up with a much better color scheme than me here!
- The images under "Donate to the election fund" by the big donate button has a white background which looks bad in dark mode. It seems I can't export them from figma without the background, but maybe you have some different source files for these with a transparent background?
- How do we want to display the timeline on mobile? For now, it just gets super-squished but I'm sure there's something nicer we can do here.

**Comment by oetherington** [comment]
Also, if you want to see the "you are here" text in the timeline, you can change line 110 of `Timeline.tsx` from 
```
  const currentDate = useCurrentTime();
```
to, for instance,
```
  const currentDate = new Date("2023-11-18");
```

**Comment by agnesstenlund** [comment]
wowwwww that was fast, and looks really good 🥲  I'll probably make some tweaks only because I'm making tweaks to the designs. 

Only "bug" I found at this first look was that the left nav is still showing 
Screenshot 2023-10-19 at 19 15 29

File: packages/lesswrong/components/ea-forum/giving-portal/EAGivingPortalPage.tsx [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+      end: new Date("2023-12-07"),
+      description: "Marginal Funding Week",
+    },
+    {
+      start: new Date("2023-12-08"),
+      end: new Date("2023-12-16"),
+      description: "Forum BOTEC-a-thon Week",
+    },
+  ],
+};
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/giving-portal/EAGivingPortalPage.tsx:214]: These timeline dates are off by one for me:

Screen Shot 2023-10-23 at 6 08 24 PM

File: packages/lesswrong/components/ea-forum/giving-portal/ElectionFundCTA.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        <div className={classes.h2}>
+          ${amountRaised} raised so far
+        </div>
+        <div className={classes.h3}>
+          Voting opens in {daysUntilVotingOpens} days
+        </div>
+      </div>
+      <button onClick={onDonate} className={classes.donateButton}>
+        Donate
+      </button>
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/giving-portal/ElectionFundCTA.tsx:109]: I think our convention is to darken buttons on hover - I noticed that this one gets lighter on hover though

**Review by s-cheng — APPROVED** [review]
Looks really nice! Since there's not much functionality and the UI isn't complete I just did a quick pass and assumed that @agnesstenlund would catch any UI issues.

**Comment by oetherington** [comment]
Copying from the conversation on Slack: there's still a couple of bugs here but we've tracked them in Slack/Asana. We're merging this into the feature branch now so we can get started on other related tasks.

**Pull request merged**